interface/orders/single_order_results.inc.php

```diff
@@ -138,13 +138,6 @@ function generate_result_row(&$ctx, &$row, &$rrow, $priors_omitted = false)
     // Have to make sure results are consistent with those before that change.
     $result_comments = str_replace("\r", "\n", $result_comments);
 
-    if ($i = strpos($result_comments, "\n")) { // "=" is not a mistake!
-    // If the first line of comments is not empty, then it is actually a long textual
-    // result value with lines delimited by "~" characters.
-        $result_comments = str_replace("~", "\n", substr($result_comments, 0, $i)) .
-        substr($result_comments, $i);
-    }
```

Comment:
A lot of assumptions here. Depending on the lab '\r' or '\n' or '\r\n' can be sent to separate notes. This script assumes '/n' looks like. Need to test for all and replace with '\r'.
Replacing the tilde with '\n' is important because if the note isn't formatted it may not make sense by displaying as one long string. Screen shot may give you a reference of what some result notes can look like where I've seen test return 10+ notes. Some long and some one liners.
![image](https://user-images.githubusercontent.com/1859985/139568297-a9a2f1bb-2ce2-4880-98a0-ffc0cbe07554.png)

Reply:
thanks @sjpadgett, that happens first thing in [receive_hl7_results](https://github.com/openemr/openemr/pull/4702/files#diff-371c548b40fa2e4758f14c5aad6bc68c2a6391900c0a3721d555f33a10336ed6L731-L732)

Reply:
have you looked at one of their obx(I think) segments to see what they send for end of line? I'm pretty sure this is same script I used and put in code so, something is off.
I know this seems minor but the lab notes extend the results with sometimes important info to order author.

Reply:
![Screenshot from 2021-10-31 11-06-45](https://user-images.githubusercontent.com/5618372/139590066-4ad7dc49-c50d-47d2-9258-9f529e9a5a6e.png)

Reply:
```
MSH|^~\&|VITLTRANS|CVMC|CMSOEMR||20210924150008||ORU^R01|8082146|P|2.3|||AL
PID|1||0700014683||VITLEPICTEST^PATONE2||20001021|M||||||||||40099461c
PV1||I|^^^CVMC|||||||||||||||||||||||||||||||||||||||||20210716113500
OBR|1|111698217c-2|111698217c|2^Consults|||20210719104701|||||||||1881648418^KELLOGG^ROGER||||||20210719104701||TRANS|F|||1548374382^DIER^DOUGLAS
OBX|1|TX|2^Consults
OBX|2|TX|2^Consults
OBX|3|TX|2^Consults||CVMC: 
OBX|4|TX|2^Consults||Inpatient Consult Note
OBX|5|TX|2^Consults
OBX|6|TX|2^Consults||Inpatient Admit Date: 7/16/21
OBX|7|TX|2^Consults||Initial Observation Date (if different): 7/16/2021
OBX|8|TX|2^Consults||Date of Service: 7/19/2021
OBX|9|TX|2^Consults||Requesting Physician:
OBX|10|TX|2^Consults||Specialty Completing Consult:
OBX|11|TX|2^Consults||Reason for Consult:
OBX|12|TX|2^Consults
OBX|13|TX|2^Consults||HPI:(include onset,location,quality,severity, duration, timing, associating
OBX|14|TX|2^Consults||symptoms)
OBX|15|TX|2^Consults
OBX|16|TX|2^Consults
OBX|17|TX|2^Consults||PMH PSH
OBX|18|TX|2^Consults||No past medical history on file.
OBX|19|TX|2^Consults|| No past surgical history on file.
OBX|20|TX|2^Consults
OBX|21|TX|2^Consults||Social History Family History
OBX|22|TX|2^Consults||Social History
OBX|23|TX|2^Consults
OBX|24|TX|2^Consults||Tobacco Use
OBX|25|TX|2^Consults||~      Smoking status: Not on file
OBX|26|TX|2^Consults||Substance Use Topics
OBX|27|TX|2^Consults||~      Alcohol use: Not on file
OBX|28|TX|2^Consults|| No family history on file.
OBX|29|TX|2^Consults
OBX|30|TX|2^Consults
OBX|31|TX|2^Consults
OBX|32|TX|2^Consults||Medications
OBX|33|TX|2^Consults||Current Facility-Administered Medications
OBX|34|TX|2^Consults||Medication Route Frequency
OBX|35|TX|2^Consults||~      acetaminophen (TYLENOL) tablet 650 mg oral Q6H PRN
OBX|36|TX|2^Consults||~      oxyCODONE immediate release tablet 5 mg oral ONCE
OBX|37|TX|2^Consults||~      simvastatin (ZOCOR) tablet 20 mg oral QPM
OBX|38|TX|2^Consults
OBX|39|TX|2^Consults
OBX|40|TX|2^Consults||Allergies
OBX|41|TX|2^Consults||No Known Allergies
OBX|42|TX|2^Consults

Reply:
I'd be willing to bet that there is not a '\r' separating orders. All I can say is the tilde's need to be replaced with '\n' and not simply parsed out.

If you send me a result file, I'll parse and look at..

Reply:
the separator in this case is `CR` or hex `D`

Reply:
[epic.zip](https://github.com/openemr/openemr/files/7449181/epic.zip)

Reply:
looks like am not picking up `Note 3`, `Performing Organization`, will try to track down

Reply:
works for me but i'm on window 11 so... Also tilde replace is working and lastly, this labs comment formatting is horrid(Halloween pun)
![image](https://user-images.githubusercontent.com/1859985/139595206-bf484a8e-ad02-4965-bff8-9c6122591ffc.png)

Reply:
What is an example result file name giving you issues.

Reply:
Recommend:
$result_comments = str_replace("~", "\r\n", substr($result_comments, 0, $i))

Reply:
sorry, too much jumping around to diff reports, `Note 1` shows up fine, mean the `CBC` you just posted, no 1 or 2 in the Note column, `8076949.hl7`

Reply:
> Recommend: $result_comments = str_replace("~", "\r\n", substr($result_comments, 0, $i))

won't that remove the squigglies that am trying to display?

Reply:
Thats what it's for. You don't want tildes in the comment. Tildes represent a CR in comments because you don't send CR in HL7 messages except to end a message.

Reply:
am supposed to show the tildes
```
OBX|35|TX|2^Consults||~      acetaminophen (TYLENOL) tablet 650 mg oral Q6H PRN
```

Reply:
I see your issue with the NTE segments not recording the comment ID however I don't think it needs to or where would it go anyway. I think only OBX get an id for a result comment.

Reply:
> am supposed to show the tildes

HUH! Give me a call as I've never seen this out of at the two major labs and 5 smaller labs!

Reply:
yeah, weirdness, if it's supposed to be displayed they should be sending `\R\` Right?

Reply:
I don't know how to help Stephen for as far as I'm concerned, this is working as it's supposed to. I still say our report is well done and has been certified by several labs. I'd push back on your lab because they can't really tell you how to format our report to user except that the info is presented! 
Besides, if you change how the tilde is used you'll need to put in a conditional for this lab so other labs reports remain the same.